linkerd/stack/src/proxy.rs

```diff
@@ -0,0 +1,87 @@
+use futures::{future, Future, Poll};
+use linkerd2_error::Error;
+
+/// A middleware type that cannot exert backpressure.
+///
+/// Typically used to modify requests or responses.
+pub trait Proxy<Req, S: tower::Service<Self::Request>> {
```

Comment:
This is, admittedly, extremely bikesheddy, but I'm not a huge fan of calling this a `Proxy`...my immediate assumption upon seeing something called a `Proxy` is that it's bundling together all the behavior of a inbound or outbound side of the proxy, or an entire linkerd instance.

I'm not sure what to suggest as an alternative though. My first thought on reading "A middleware type that cannot exert backpressure" is `ReadyService`, but that's been used in the past to mean something different...it's basically like a `map` operation for requests/responses, but I don't know if it makes sense to call the trait `Map`...I almost want to suggest `Filter`...

I'm probably overthinking it, and the name `Proxy` is fine...

Reply:
Is `ServiceProxy` any better?

Reply:
wordier but clearer, IMO!

Reply:
Then is the service it produces `ServiceProxyService`?

Reply:
I kind of lean towards the original `Proxy`, but definitely not a blocker for `ServiceProxy`

Reply:
`ServiceProxy` is the trait, `ProxyService` is the `Service`, obviously! :P